embedding/cpp/include/flutter_engine.h

```diff
@@ -0,0 +1,112 @@
+// Copyright 2022 Samsung Electronics Co., Ltd. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_TIZEN_EMBEDDING_CPP_INCLUDE_FLUTTER_ENGINE_H_
+#define FLUTTER_TIZEN_EMBEDDING_CPP_INCLUDE_FLUTTER_ENGINE_H_
+
+#include <app_control.h>
+#include <flutter/plugin_registry.h>
+#include <flutter_tizen.h>
+
+#include <memory>
+#include <optional>
+#include <string>
+#include <vector>
+
+// The engine for flutter execution.
+class FlutterEngine : public flutter::PluginRegistry {
+ public:
+  virtual ~FlutterEngine();
+  static std::unique_ptr<FlutterEngine> Create(
+      const std::optional<std::string>& dart_entrypoint = std::nullopt,
+      const std::optional<std::vector<std::string>>& dart_entrypoint_args =
+          std::nullopt);
```

Comment:
The use of `std::optional` seems redundant since an empty `dart_entrypoint` already indicates that the value will default to `main` ([doc](https://github.com/flutter-tizen/flutter-tizen/blob/47918330dd65e407c988f91349f16a71ef5217cb/embedding/cpp/include/flutter_app.h#L80-L81)). One option is to just revert b89fdd08e116430d41260369abbe9eb282b70a54. The other options will involve significant changes to the embedder API so we might consider them later. One of such options is to pass `FlutterDesktopEngineProperties` on `EngineRun` (instead of `EngineCreate`), but in that case we need a way to properly deal with logging options (`--verbose-logging` and `--tizen-logging-port`).

Reply:
I replaced it with the default argument.